src/mlpack/methods/gmm/em_fit_impl.hpp

```diff
@@ -373,21 +383,23 @@ LogLikelihood(const arma::mat& observations,
 {
   double logLikelihood = 0;
 
-  arma::vec phis;
-  arma::mat likelihoods(dists.size(), observations.n_cols);
+  arma::vec logPhis;
+  arma::mat logLikelihoods(dists.size(), observations.n_cols);
 
+  // It has to be LogProbability() otherwise Probability() would overflow easily
   for (size_t i = 0; i < dists.size(); ++i)
   {
-    dists[i].Probability(observations, phis);
-    likelihoods.row(i) = weights(i) * trans(phis);
+    dists[i].LogProbability(observations, logPhis);
+    logLikelihoods.row(i) = log(weights(i)) + trans(logPhis);
   }
   // Now sum over every point.
   for (size_t j = 0; j < observations.n_cols; ++j)
   {
-    if (accu(likelihoods.col(j)) == 0)
+    if (mlpack::math::AccuLog(logLikelihoods.col(j)) ==
+                                      -std::numeric_limits<double>::infinity())
```

Comment:
That looks strange as well, I would probably adapt the itendation here as well (tab twice, 4 spaces).